misc/toml-parser/src/main/java/org/ballerinalang/toml/parser/ManifestBuildListener.java

```diff
@@ -0,0 +1,258 @@
+/*
+ *  Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+package org.ballerinalang.toml.parser;
+
+import org.ballerinalang.toml.antlr4.TomlBaseListener;
+import org.ballerinalang.toml.antlr4.TomlParser;
+import org.ballerinalang.toml.model.Dependency;
+import org.ballerinalang.toml.model.Manifest;
+import org.ballerinalang.toml.model.fields.DependencyField;
+import org.ballerinalang.toml.model.fields.ManifestHeader;
+import org.ballerinalang.toml.model.fields.PackageField;
+import org.ballerinalang.toml.util.SingletonStack;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Custom listener which is extended from the Toml listener with our own custom logic.
+ *
+ * @since 0.964
+ */
+public class ManifestBuildListener extends TomlBaseListener {
+    private final Manifest manifest;
+    private final SingletonStack<String> currentKey = new SingletonStack<>();
+    private Dependency dependency;
+    private String currentHeader = null;
+
+    /**
+     * Cosntructor with the manifest object.
```

Comment:
Typo?

Reply:
Removed all the custom listeners from the toml-parser